MHV-62950: Success/error message still shows when navigating back to Refill page

vmenshutin-bylight opened the pull request:
## Summary

Refill notification data is now cleared from Redux store on Refill page load.

## Related issue(s)

https://jira.devops.va.gov/browse/MHV-62950

<img width="950" alt="image" src="https://github.com/user-attachments/assets/4047048e-989f-4102-9e3d-86fee8a14560">

## Testing done

- Manual testing

## What areas of the site does it impact?

/my-health/medications

## Acceptance criteria

AC1: When navigating back to the refill page after user submits a refill then goes to the list page and back to refill page, the alert should no longer be there. 

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

vmenshutin-bylight committed:
MHV-62950: Success/error message still shows when navigating back to refill page

robertbylight commented on src/applications/mhv-medications/actions/prescriptions.js:
```diff
@@ -108,3 +108,7 @@ export const fillPrescriptions = prescriptions => async dispatch => {
   }
 };
+
+export const clearFillNotification = async dispatch => {
```
would you be able to add a test case for this function?

vmenshutin-bylight replied:
TY. done!

robertbylight commented on src/applications/mhv-medications/reducers/prescriptions.js:
```diff
@@ -106,6 +106,12 @@ export const prescriptionsReducer = (state = initialState, action) => {
         apiError: false,
       };
     }
+    case Actions.Prescriptions.CLEAR_FILL_NOTIFICATION: {
```
would you be able to add a test case for this as well?

vmenshutin-bylight replied:
TY. done!

agravell047 commented on src/applications/mhv-medications/containers/RefillPrescriptions.jsx:
```diff
@@ -117,6 +118,7 @@ const RefillPrescriptions = ({ isLoadingList = true }) => {
   };
 
   useEffect(() => {
+    dispatch(clearFillNotification);
```
probably not much of a performance increase but you could add a check to clear only if there's an existing notification

vmenshutin-bylight replied:
TY. Done!

vmenshutin-bylight committed:
addressed PR comments

Commit:
e2e test for no alert on refill page

tests added for navigating back to refill page after success ,failed , and partial requests
